### PR TITLE
Add Main Page for ADB calculator

### PR DESCRIPTION
**Pull request by wopni1021**
image

Demo video:
https://app.screencastify.com/v3/watch/aieQI9Rz9C3M2o52cqFD

### CODE REVIEW

File: my-app/src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
+import Table from "./components/MainTable";
+import Tab from "./components/Tab";
+import Feedback from "./components/Feedback";
+import Header from "./components/Header";
+const App = () => {
+  const [selectedTab, setSelectedTab] = useState(0);
+
+  const handleChangeTab = (newValue: number) => {
+    setSelectedTab(newValue);
+  };
```
Comment (zhenshuo) [inline my-app/src/App.tsx:12]: useCallback

File: my-app/src/App.tsx [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
+
+  const renderPage = () => {
+    switch (selectedTab) {
+      case 0:
+        return <Table />;
+      case 1:
+        return null;
+      case 2:
+        return <Feedback />;
+    }
```
Comment (zhenshuo) [inline my-app/src/App.tsx:22]: missing default case

File: my-app/src/components/Feedback.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import "./Feedback.scss";
+
+const ROOT = "adb-feedback";
```
Comment (zhenshuo) [inline my-app/src/components/Feedback.tsx:4]: unnecessary variable

File: my-app/src/components/MainTable.tsx [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+  return /^\d*\.?\d{0,2}$/.test(value);
+};
+
+const MainTable = () => {
+  const firstDate = dayjs().date(1);
+  const [rows, setRows] = useState<State>(initialRows);
+  const [initialBalance, setInitialBalance] = useState<number>(0);
+  const [initialDate, setInitialDate] = useState<Dayjs | null>(firstDate);
+
+  const handleChangeWithDr = (value: string, index: number) => {
```
Comment (zhenshuo) [inline my-app/src/components/MainTable.tsx:105]: useCallback

File: my-app/src/components/MainTable.tsx [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+  };
+
+  const onClickAddTransaction = (index: number) => {
+    setRows((prevRows) => {
+      const parentRow = prevRows[index];
+      const childRow = { ...parentRow, withdraw: 0, depo: 0 };
+      const newState = [...prevRows.slice(0, index + 1), childRow, ...prevRows.slice(index + 1)];
+      return newState;
+    });
+  };
```
Comment (zhenshuo) [inline my-app/src/components/MainTable.tsx:189]:
```
setRows(prevRows => prevRows.map((row,idx) => {
  if (idx === index) {
    return { 
      ...row, withdraw: 0, depo: 0 
    }; 
  } 
  return row;
});
```

File: my-app/src/components/MainTable.tsx [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+        </StyledTableCell>
+        <StyledTableCell align="right">
+          <TextField
+            {...commonInputProps}
+            id="depo"
+            variant="outlined"
+            type="number"
+            onChange={(e) => {
+              handleChangeDepo(e.target.value, index);
+            }}
```
Comment (zhenshuo) [inline my-app/src/components/MainTable.tsx:237]: create a memorized function for the handler

File: my-app/package.json [hunk 7]
```diff
@@ -3,16 +3,24 @@
...
     "@types/react": "^18.3.4",
     "@types/react-dom": "^18.3.0",
+    "dayjs": "^1.11.13",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
     "react-scripts": "5.0.1",
+    "sass": "^1.78.0",
     "typescript": "^4.9.5",
     "web-vitals": "^2.1.4"
   },
```
Comment (zhenshuo) [inline my-app/package.json:26]: add  https://prettier.io/docs/en/integrating-with-linters.html
```
devDependencies: {
  "prettier"
   "eslint-config-prettier"
  "eslint-plugin-prettier"
}
```

File: my-app/src/components/Feedback.tsx (hunk 3 above)
Comment (wopni1021) [reply my-app/src/components/Feedback.tsx:4]: Used as classnames

File: my-app/src/components/MainTable.tsx (hunk 5 above)
Comment (wopni1021) [reply my-app/src/components/MainTable.tsx:189]: It was trying to insert a new row in the middle instead of replacing an existing row.

**Pull request merged**